psh/pm/pm.c

```diff
@@ -104,9 +139,30 @@ int psh_pm(int argc, char *argv[])
 				interval = min(1, atoi(optarg));
 				break;
 
+			case 'm':
+				maxTotal = strtoul(optarg, &str, 10);
+				if (str == optarg) {
+					fprintf(stderr, "pm: invalid -m value\n");
+					return EXIT_FAILURE;
+				}
+				printf("pm: monitoring total mem usage - limit %u bytes\n", maxTotal);
+				rebootNoMem = 1; /* monitoring memory - reboot also when we encounter ENOMEM error internally */
+				break;
+
+			case 'k':
+				maxKernel = strtoul(optarg, &str, 10);
+				if (str == optarg) {
```

Comment:
The below suggestion may help to filter invalid optarg value e.g. `-k 123abcd`
```suggestion
				if (str == optarg || *str != '\0') {
```